annotation_tool.py

```diff
@@ -0,0 +1,245 @@
+import datetime
+import json
+import os
+
+import fiona
+import ipywidgets
+import leafmap
+import solara
+
+from leafmap import toolbar
+
+#
+# Application variables
+#
+# Map variables
+center = solara.reactive((20, 0))
+zoom = solara.reactive(4)
+
+# STAC creation variables
+stac_annotation_filename = solara.reactive("data.geojson")
+stac_asset_link = solara.reactive("")
+stac_asset_title = solara.reactive("")
+stac_item_link = solara.reactive("")
+stac_collection_id = solara.reactive("")
+stac_file_name = solara.reactive("")
+stac_item_id = solara.reactive("")
+
+# Behavior variables
+continuous_update = solara.reactive(True)
+show_instruction_message = solara.reactive(True)
+
+
+#
+# Utility functions
+#
+def get_annotation_bbox(filename=stac_annotation_filename.value):
+    """
+    Create bounding box for a given annotation file
+
+    Args:
+        filename: File name of the geojson annotation file
+
+    Returns: Bounding box in list form
+
+    """
+    annotations = fiona.open(filename)
+    return list(annotations.bounds)
+
+
+def create_stac_item(
+    filename=stac_annotation_filename.value,
+    stac_id=stac_item_id.value,
+    stac_title=stac_asset_title.value,
+    stac_asset=stac_asset_link.value,
+    collection_id=stac_collection_id.value,
+    item_link=stac_item_link.value,
+):
+    """
+    Create a STAC item stub from a geojson annotation file, which
+    can later be expanded.
+
+    Args:
+        filename: Geojson annotation file name/path
+        stac_id: ID for new STAC item
+        stac_asset: Link to the asset, which is the hosted Geojson annotation file
+        stac_title: Title for the Geojson annotation file asset
+        collection_id: STAC collection ID, if applicable
+        item_link: STAC asset link of EO data used to create annotation
+
+    """
+    bbox = get_annotation_bbox(filename)
+    file_size = os.stat(filename).st_size
+    item = {
+        "stac_version": "1.0.0",
+        "type": "Feature",
+        "id": stac_id,
+        "bbox": bbox,
+        "geometry": {"type": "Polygon", "coordinates": bbox},
+        "properties": {"created": datetime.datetime.now().isoformat()},
```

Comment:
For labeling, should include the following:
(using YAML for readability, but port to equivalent JSON)

```yaml
label:properties: 
  - "<class-field>"  # name of the annotation (usually only "class", but usefull if more than one annotation per sample is provided)
label:tasks: 
  # depending on what applies 1-or-more (since using bbox, probably just "classification" applies without further input knowledge [?])
  - "classification"
  - "detection"     # if the item annotated is some object (car, plane, etc.) not just plain land-cover forest, sea, etc.
  - "segmentation"  # if the annotation was surrounding the specific object, typically not just a simple bbox/polygon
  - "regression"    # if there is some floating value associated, such as "building condition", "pollution level", etc.
label:methods:
  - manual  # for this case
label:type: "vector"  # for a GeoJSON annotation (if labels were a GeoTiff, eg: a mask, it would be "raster")
label:description: "..."  # optional, but could refer to this app to indicate how annotation was done
label:classes:
  - name: "<class-field>"  # typically simply "class", should match contents of "label:properties"
    classes: [ "<class-name>", "<class-index>" ]  # all possible representations to refer to the class in a dataset
label:overviews:
  # stats about the annotations
  - property_key: "<class-field>"
    counts:
      # for each <class-name>...
      - name: "<class-name>" or <class-index> (as string)
        count: <number of occurences>
```

Reply:
Should be ok now